### PR TITLE
Modernize CVMFS tests

### PR DESCRIPTION
**Pull request by brianhlin**
A bunch of helpful stuff that was added to debug the CVMFS EL6 failures, More or less the same as what ran here minus a few specific debug commits: https://osg-sw-submit.chtc.wisc.edu/tests/20201015-1225/results.html.

### CODE REVIEW

File: osgtest/tests/test_180_cvmfs.py [hunk 1]
```diff
@@ -2,62 +2,61 @@
...
-        contents=[]
-        contents.append("CVMFS_SERVER_URL=\"http://cvmfs.fnal.gov:8000/opt/@org@;http://cvmfs.racf.bnl.gov:8000/opt/@org@;http://cvmfs-stratum-one.cern.ch:8000/opt/@org@;http://cernvmfs.gridpp.rl.ac.uk:8000/opt/@org@\"\n")
-        files.write("/etc/cvmfs/domain.d/cern.ch.local", contents, owner='cvmfs', chmod=0o644)
-
-
-    def test_01_setup_cvmfs(self):
-        core.skip_ok_unless_installed('cvmfs')
+CVMFS_CONFIG = """CVMFS_REPOSITORIES="`echo $((echo oasis.opensciencegrid.org;echo cms.cern.ch;ls /cvmfs)|sort -u)|tr ' ' ,`"
+CVMFS_QUOTA_LIMIT=10000
+CVMFS_HTTP_PROXY="http://squid-cs-b240.chtc.wisc.edu:3128|http://squid-cs-2360.chtc.wisc.edu:3128|http://squid-wid.chtc.wisc.edu:3128;DIRECT"
```
Comment (matyasselmeci) [inline osgtest/tests/test_180_cvmfs.py:7]: should this be a `|DIRECT` instead of `;DIRECT`?

File: osgtest/tests/test_180_cvmfs.py [hunk 2]
```diff
@@ -2,62 +2,61 @@
...
 
-        stdout, stderr, fail = core.check_system(('service', 'autofs', 'restart'), 'Start cvmfs server')
-        self.assert_(stdout.find('FAILED') == -1, fail)
-        core.state['cvmfs.started-server'] = True
+        setup_fuse()
+        setup_automount()
+        setup_cvmfs()
 
+        stdout, _, fail = core.check_system(('service', 'autofs', 'restart'), 'Start cvmfs server')
+        self.assertEqual(stdout.find('FAILED'), -1, fail)
```
Comment (matyasselmeci) [inline osgtest/tests/test_180_cvmfs.py:61]:
```suggestion
        self.assertFalse("FAILED" in stdout, fail)
```

File: osgtest/tests/test_180_cvmfs.py (hunk 1 above)
Comment (brianhlin) [reply osgtest/tests/test_180_cvmfs.py:7]: Nope, from https://opensciencegrid.org/docs/worker-node/install-cvmfs/#configuring-cvmfs:

> A semicolon between proxies means to try that one only after the previous ones have failed.

File: osgtest/tests/test_380_cvmfs.py [hunk 3]
```diff
@@ -14,9 +17,22 @@ def debug_cvmfs(self, repo):
...
-        core.check_system(command, 'Manual cvmfs mount failed')
-        # If manual mount works, autofs is broken
-        self.fail("Autofs failed to mount /cvmfs/%s" % repo)
+        status, _, _ = core.system(command)
+
+        # If manual mount works, autofs is likely the culprit
+        if status:
+            debug_contents = "Failed to manually mount %s\n" % repo
+        else:
+            debug_contents = "Successful manual mount of %s" % repo
```
Comment (matyasselmeci) [inline osgtest/tests/test_380_cvmfs.py:26]:
```suggestion
            debug_contents = "Successful manual mount of %s\n" % repo
```

File: osgtest/tests/test_380_cvmfs.py [hunk 4]
```diff
@@ -14,9 +17,22 @@ def debug_cvmfs(self, repo):
...
+
+        debug_file = "/tmp/cvmfs_debug.log"
+        debug_contents += '='*20
```
Comment (matyasselmeci) [inline osgtest/tests/test_380_cvmfs.py:29]:
```suggestion
        debug_contents += ('='*20) + "\n"
```

**Review by matyasselmeci — APPROVED** [review]
Few minor changes; pre-approving.

**Pull request merged**